presentation-compiler/src/main/dotty/tools/pc/MetalsInteractive.scala

```diff
@@ -0,0 +1,341 @@
+package dotty.tools.pc
+
+import scala.annotation.tailrec
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.ast.untpd
+import dotty.tools.dotc.core.ContextOps.*
+import dotty.tools.dotc.core.Contexts.*
+import dotty.tools.dotc.core.CyclicReference
+import dotty.tools.dotc.core.Flags.*
+import dotty.tools.dotc.core.Names.Name
+import dotty.tools.dotc.core.StdNames
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.dotc.core.Types.Type
+import dotty.tools.dotc.interactive.SourceTree
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+
+object MetalsInteractive:
+
+  // This is a copy of dotty.tools.dotc.Interactive.contextOfPath
+  // with a minor fix in how it processes `Template`
```

Comment:
We could probably apply the fix now instead of copying

Reply:
Can be next PR, though this should be pretty straightforward, since this was done as a workaround

Reply:
> I haven't applied any changes to the metals, that weren't required. Do you want me to do this in this PR on separate one ?

I would personally prefer to have this in this PR just so we don't forget about it :).